lib/dissoc/index.js

```diff
@@ -0,0 +1,8 @@
+import _curry2 from '../_internal/_curry2'
+
+export default _curry2(function dissoc(prop, obj) {
+  var res = {}
+  for (var key in obj) res[key] = obj[key]
+  delete res[prop]
```

Comment:
as I know `delete` keyword doesn't allow optimisation of browsers to shorten the code or make the code faster, maybe replace with `res[prop] = undefined` or is it not what we want to?

Reply:
`res[prop] = undefined` will not delete prop, so in functional paradigm, this method will not omit prop, it would make it `undefined`

```js
const a = { a: 1, b: 2 }
a.a = undefined
// a - { a: undefined, b: 2 }

delete a.a
// a - { b: 2 }
```

Reply:
Also, we could not use `delete`, but check `if (key !== path[0]) res[key] = obj[key]` in every loop iteration. But it's a question: would it be faster ?

Reply:
maybe `Object.keys(obj).filter(function(p) { return p !== prop })`
and then your `5`th line
Yeah, I know it's so slow at first sight

Reply:
It's a question for discussion. Do you agree, we should do it later, because we could spend a lot of time for researching?